# Liz Young - Code Review - Technical Interview Presentation

<hr>

In this notebook I will walk through code I wrote to start analyzing reviews of musical instruments from Amazon purchases.

I selected [this dataset](https://www.kaggle.com/datasets/eswarchandt/amazon-music-reviews?resource=download) from Kaggle, after searching for semistructured and unstructured datasets, this one resonated with me.

<hr>

I set up a private github repo tracking my progress with this project, you can find via [this link](https://github.com/completelyAbsorbed/amazon-musical-instrument-reviews), containing ...
- commits of this project from start to finish
- a progress tracker document I designed and used to execute the PACE methodology on this project
- screen captures of my git commits demonstrating competency with git, github, and commandline
- the original database sourced from Kaggle
- this Jupyter notebook
- any files produced by this notebook

<hr>

### *Thank you for your time and consideration*

<hr>

# Import

In [ ]:
# import packages and libraries
import pandas as pd

# Load Original Data

In [7]:
# define the filename where python can find the original dataset
data_filename = "Musical_instruments_reviews.csv"

# read in the original dataset to a pandas DataFrame
data_raw = pd.read_csv(data_filename)

# Initial Inspection of Data

<hr>

*Learn more about the data using methods from the pandas DataFrame type. <br> <br>
Demonstrating competency with software design pattern: Object-Oriented Programming.*

In [9]:
# look at the first 5 rows of the original dataset
print(data_raw.head())

       reviewerID        asin  \
0  A2IBPI20UZIR0U  1384719342   
1  A14VAT5EAX3D9S  1384719342   
2  A195EZSQDW3E21  1384719342   
3  A2C00NNG1ZQQG2  1384719342   
4   A94QU4C90B1AX  1384719342   

                                       reviewerName   helpful  \
0  cassandra tu "Yeah, well, that's just like, u...    [0, 0]   
1                                              Jake  [13, 14]   
2                     Rick Bennette "Rick Bennette"    [1, 1]   
3                         RustyBill "Sunday Rocker"    [0, 0]   
4                                     SEAN MASLANKA    [0, 0]   

                                          reviewText  overall  \
0  Not much to write about here, but it does exac...      5.0   
1  The product does exactly as it should and is q...      5.0   
2  The primary job of this device is to block the...      5.0   
3  Nice windscreen protects my MXL mic and preven...      5.0   
4  This pop filter is great. It looks and perform...      5.0   

                   

In [10]:
# look at the last 5 rows of the original dataset
print(data_raw.tail())

           reviewerID        asin             reviewerName helpful  \
10256  A14B2YH83ZXMPP  B00JBIVXGC          Lonnie M. Adams  [0, 0]   
10257   A1RPTVW5VEOSI  B00JBIVXGC       Michael J. Edelman  [0, 0]   
10258   AWCJ12KBO5VII  B00JBIVXGC         Michael L. Knapp  [0, 0]   
10259  A2Z7S8B5U4PAKJ  B00JBIVXGC  Rick Langdon "Scriptor"  [0, 0]   
10260  A2WA8TDCTGUADI  B00JBIVXGC          TheTerrorBeyond  [0, 0]   

                                              reviewText  overall  \
10256            Great, just as expected.  Thank to all.      5.0   
10257  I've been thinking about trying the Nanoweb st...      5.0   
10258  I have tried coated strings in the past ( incl...      4.0   
10259  Well, MADE by Elixir and DEVELOPED with Taylor...      4.0   
10260  These strings are really quite good, but I wou...      4.0   

                                                 summary  unixReviewTime  \
10256                                         Five Stars      1405814400   
10257  Long 

In [15]:
# print a summary of the size, non-null counts, and column variable types of the original dataset
print(data_raw.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10261 entries, 0 to 10260
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   reviewerID      10261 non-null  object 
 1   asin            10261 non-null  object 
 2   reviewerName    10234 non-null  object 
 3   helpful         10261 non-null  object 
 4   reviewText      10254 non-null  object 
 5   overall         10261 non-null  float64
 6   summary         10261 non-null  object 
 7   unixReviewTime  10261 non-null  int64  
 8   reviewTime      10261 non-null  object 
dtypes: float64(1), int64(1), object(7)
memory usage: 721.6+ KB
None


The original dataset is 9 columns and 10261 rows.

The columns provided are ... 
> 'reviewerID'
    <br>    'asin'
    <br>    'reviewerName'
    <br>    'helpful'
    <br>    'reviewText'
    <br>    'overall'
    <br>    'summary'
    <br>    'unixReviewTime'
    <br>    'reviewTime'
    
The original dataset contains no Null values.

Column 'overall' is type float64, 'unixReviewTime' is type int64, the rest are type object.

The original dataset uses ~721.6 KB in memory.


In [19]:
# examine count, mean, standard deviation, minimum, maximum, and quartiles of numeric columns in the original dataset
print(data_raw.describe())

            overall  unixReviewTime
count  10261.000000    1.026100e+04
mean       4.488744    1.360606e+09
std        0.894642    3.779735e+07
min        1.000000    1.095466e+09
25%        4.000000    1.343434e+09
50%        5.000000    1.368490e+09
75%        5.000000    1.388966e+09
max        5.000000    1.405987e+09


'overall' is the reviewers score per the [data dictionary](https://www.kaggle.com/datasets/eswarchandt/amazon-music-reviews?resource=download). It ranges from 1 to 5.

'unixReviewTime' represents unix time of the review, and is directly related to 'reviewTime' (not numeric, so not summarised here).

Next, it would be helpful to inspect each of the non-numeric columns, and ascertain their features and usefulness.